File: Curso-Logica-Unicred/Python/comerciante.py

nome: str

quantBetewn = 0
quant10 = 0
quant20 = 0

porcento = 0.00
compra = 0.00
venda = 0.00
totalCompra = 0.00
totalVenda = 0.00

quantidadeProduto = int(input("Serao digitados dados de quantos produtos? "))

for i in range(0, quantidadeProduto):

    print(f"Produto {i+1}: ")

    nome = input("Nome: ")

    compra = float(input("Preço de compra: "))

    venda = float(input("Preço de venda: "))

    porcento = (( venda - compra) / compra * 100)

    totalCompra = totalCompra + compra
    totalVenda = totalVenda + venda

    if (porcento < 10.0):
        quant10 = quant10+1
    elif (porcento < 20.0):
        quantBetewn = quantBetewn+1
    else: 
        quant20 = quant20+1


print(" RELATORIO: ")
print(f"Lucro abaixo de 10%: {quant10}")
print(f"Lucro entre 10% e 20%: {quantBetewn}")
print(f"Lucro acima de 20%: {quant20}")
print(f"Valor total de compra: {totalCompra:.2f}")
print(f"Valor total de venda: {totalVenda:.2f}")
print(f"Lucro total: {(totalVenda - totalCompra):.2f}")

File: Curso-Logica-Unicred/C/comerciante.c
#include <stdio.h>
#include <string.h>

void limpar_entrada() {
    char c;
    while ((c = getchar()) != '\n' && c != EOF) {}
}

void main() {
    int quantidadeProduto, quant10, quant20, quantBetewn;
    char nome[50];
    double porcento, compra, venda, totalCompra, totalVenda;

    printf("Serao digitados dados de quantos produtos? ");
    scanf("%d", &quantidadeProduto);

    for (int i = 0; i < quantidadeProduto; i++) {

        printf("Produto %d: \n", i+1);

        printf("Nome: ");
        limpar_entrada();
        scanf("%s", nome);

        printf("Preço de compra: ");
        scanf("%lf", &compra);

        printf("Preço de venda: ");
        scanf("%lf", &venda);

        porcento = (( venda - compra) / compra * 100);

        totalCompra = totalCompra + compra;
        totalVenda = totalVenda + venda;

        if (porcento < 10.0) {
            quant10++;
        } else if (porcento < 20.0) {
            quantBetewn++;
        } else {
            quant20++;
        }
    }

    printf("\n RELATORIO:\n");
    printf("Lucro abaixo de 10%%: %d \n", quant10);
    printf("Lucro entre 10%% e 20%%: %d \n", quantBetewn);
    printf("Lucro acima de 20%%: %d\n", quant20);
    printf("Valor total de compra: %.2lf \n", totalCompra);
    printf("Valor total de venda: %.2lf \n", totalVenda);
    printf("Lucro total: %.2lf \n", totalVenda - totalCompra);
}